components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/extension/eidas/EidasExtensionProcessor.java

```diff
@@ -0,0 +1,354 @@
+import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.sso.saml.SAMLSSOConstants;
+import org.wso2.carbon.identity.sso.saml.dto.SAMLSSOAuthnReqDTO;
+import org.wso2.carbon.identity.sso.saml.dto.SAMLSSOReqValidationResponseDTO;
+import org.wso2.carbon.identity.sso.saml.exception.IdentitySAML2SSOException;
+import org.wso2.carbon.identity.sso.saml.extension.SAMLExtensionProcessor;
+import org.wso2.carbon.identity.sso.saml.extension.eidas.model.RequestedAttributes;
+import org.wso2.carbon.identity.sso.saml.extension.eidas.model.SPType;
+import org.wso2.carbon.identity.sso.saml.extension.eidas.util.EidasConstants;
+import org.wso2.carbon.identity.sso.saml.util.SAMLSSOUtil;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.collections.CollectionUtils.isNotEmpty;
+
+/**
+ * This class is used to process and validate the eIDAS SAML extensions.
+ */
+public class EidasExtensionProcessor implements SAMLExtensionProcessor {
+    private static Log log = LogFactory.getLog(EidasExtensionProcessor.class);
+    private static String errorMsg = "Mandatory Attribute not found.";
+
+    /**
+     * Check whether the SAML authentication request can be handled by the eIDAS extension processor.
+     *
+     * @param request SAML request
+     * @return true if the request can be handled
+     * @throws IdentitySAML2SSOException
+     */
+    @Override
+    public boolean canHandle(RequestAbstractType request) throws IdentitySAML2SSOException {
+
+        return request.getNamespaces().stream().anyMatch(namespace -> namespace.getNamespaceURI()
+                .equals(EidasConstants.EIDAS_NS));
+    }
+
+    /**
+     * Check whether the SAML response can be handled by the eIDAS extension processor.
+     *
+     * @param authReqDTO Authentication request data object
+     * @return true if the request can be handled
+     * @throws IdentitySAML2SSOException
+     */
+    @Override
+    public boolean canHandle(StatusResponseType response, Assertion assertion, SAMLSSOAuthnReqDTO authReqDTO)
+            throws IdentitySAML2SSOException {
+
+        return authReqDTO.getProperties().getProperty(EidasConstants.EIDAS_REQUEST)
+                .equals(EidasConstants.EIDAS_PREFIX);
+    }
+
+    /**
+     * Process and Validate the SAML extensions in authentication request for EIDAS message format.
+     *
+     * @param request        Authentication request
+     * @param validationResp reqValidationResponseDTO
+     * @throws IdentityException
+     */
+    @Override
+    public void processSAMLExtensions(RequestAbstractType request, SAMLSSOReqValidationResponseDTO validationResp)
+            throws IdentitySAML2SSOException {
+
+        if (request instanceof AuthnRequest) {
+            if (log.isDebugEnabled()) {
+                log.debug("Process the extensions for EIDAS message format");
```

Comment:
This does not indicate if it's in the request flow. If SP information and user is available here mention them in debug logs